src/Tizen.NUI.WindowSystem/src/public/ScreensaverService.cs

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright(c) 2023 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.ComponentModel;
+
+namespace Tizen.NUI.WindowSystem.Shell
+{
+    /// <summary>
+    /// Class for the Tizen screensaver service.
+    /// </summary>
+    /// This class is need to be hidden as inhouse API.
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ScreensaverService : IDisposable
+    {
+        private TizenShell _tzsh;
+        private IntPtr _screensaverService;
+        private int _tzshWin;
+        private bool disposed = false;
+        private bool isDisposeQueued = false;
+
+        /// <summary>
+        /// Creates a new Screensaver Service handle.
+        /// </summary>
+        /// <param name="tzShell">The TizenShell instance.</param>
+        /// <param name="win">The window to provide service of the screensaver.</param>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when a argument is null.</exception>
+        public ScreensaverService(TizenShell tzShell, Window win)
```

Comment:
Please use `IWindowProvider` instead of `Window`. (for the sake of extensibility)

```suggestion
        public ScreensaverService(TizenShell tzShell, IWindowProvider win)
```

Refer to `KVMService` implementation.

https://github.com/Samsung/TizenFX/blob/d9b4f92c88eb73c7fa3afc93d85c55dc5d60cc19/src/Tizen.NUI.WindowSystem/src/public/KVMService.cs#L100-L117